### ISSUE TITLE
[BUG] Delombok does not remove `@With` from record components in JDK 17.0.7

### ISSUE THREAD

**Issue by mregulski** [projectlombok/lombok#3434]
**Describe the bug**
After switching from OpenJDK 17.0.5 to 17.0.7 `delombok` stopped removing `@With` annotation from record components. The appropriate method is generated correctly, but because the import is removed, delomboked records not compilable.

**To Reproduce**

Delombok a simple record with `@With` on a component:

```
//RecordAndWithTest.java
import lombok.With;

public record RecordAndWithTest(@With String aField) {
}
```

**Expected behavior**

Running delombok from Java 17.0.5 produces correct output:

```
$ java -version
openjdk version "17.0.5" 2022-10-18
OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)
OpenJDK 64-Bit Server VM Temurin-17.0.5+8 (build 17.0.5+8, mixed mode, sharing)

$ java -jar .\lombok-1.18.28.jar delombok -p .\RecordAndWithTest.java
// Generated by delombok at Tue May 30 13:19:24 CEST 2023
//RecordAndWithTest.java
public record RecordAndWithTest(String aField) {
        /**
         * @return a clone of this object, except with this updated property (returns {@code this} if an identical value is passed).
         */
        @java.lang.SuppressWarnings("all")
        public RecordAndWithTest withAField(final String aField) {
                return this.aField == aField ? this : new RecordAndWithTest(aField);
        }
}
```

I would expect the same output when runnig under JDK 17.0.7, but the delomboked file still has the annotation on the record component:

```
$ java17.0.7 -version
openjdk version "17.0.7" 2023-04-18
OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)
OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode, sharing)

$ java17.0.7 -jar .\lombok-1.18.28.jar delombok -p .\RecordAndWithTest.java
// Generated by delombok at Tue May 30 13:23:21 CEST 2023
//RecordAndWithTest.java
public record RecordAndWithTest(@With String aField) {
        /**
         * @return a clone of this object, except with this updated property (returns {@code this} if an identical value is passed).
         */
        @java.lang.SuppressWarnings("all")
        public RecordAndWithTest withAField(final String aField) {
                return this.aField == aField ? this : new RecordAndWithTest(aField);
        }
}
```

**Version info (please complete the following information):**
 - lombok: `1.18.26`, `1.18.28`
 - java: 
   - `OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)` (bug not present)
   - `OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)` (bug present)
   - `OpenJDK Runtime Environment Corretto-17.0.7.7.1 (build 17.0.7+7-LTS)` (bug present)

**Additional context**
From a quick look, a similar issue occurs with record components annotated with `@Getter`.

**Comment by Rawi01** [comment]
I fixed something similar in #3420, can you try if that also fixes this issue?

**Comment by mregulski** [comment]
It looks like it does!

With a fresh `lombok.jar` from #3420 ([commit  1599bc1](https://github.com/projectlombok/lombok/pull/3420/commits/1599bc1ce4a2440322af7a1183828c5f65dcc237)) I get the correct behaviour:
``` 
$ java17.0.7 -version
openjdk version "17.0.7" 2023-04-18
OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)
OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode, sharing)

$ java17.0.7 -jar lombok.jar delombok -p RecordAndWithTest.java
// Generated by delombok at Thu Jun 01 15:18:14 CEST 2023

//RecordAndWithTest.java
import lombok.Getter;

public record RecordAndWithTest(String aField) {
        /**
         * @return a clone of this object, except with this updated property (returns {@code this} if an identical value is passed).
         */
        @java.lang.SuppressWarnings("all")
        public RecordAndWithTest withAField(final String aField) {
                return this.aField == aField ? this : new RecordAndWithTest(aField);
        }
}
```
